File: bitwise operators/factors.java
import java.util.ArrayList;

public class factors {
    public static void main(String[] args) {
        fact2(24);
    }

    static void fact(int number) {
        for (int i = 1; i <= number; i++) {
            if (number % i == 0) {
                System.out.println(i);
            }
        }
    }

    static void fact2(int number) {
        ArrayList<Integer> nums = new ArrayList<>();
        for (int i = 1; i <= Math.sqrt(number); i++) {
            if (number % i == 0) {
                if (number / i == i) {
                    System.out.println(i);
                }

                else {
                    System.out.println(i);
                    nums.add(number / i);
                }
            }
        }

        for(int j = nums.size() - 1; j >= 0; j--) {
            System.out.println(nums.get(j));
        }
    }
}

File: bitwise operators/factors.py
import math
def factors(number):
    for i in range(1, number + 1):
        if (number % i == 0):
            print(i)

def factors2(number):
    nums = []
    for i in range(1, math.ceil(math.sqrt(number))):
        if (number % i == 0):
            if (number / i == i):
                print(i)
            else:
                print(i)
                nums.append(number // i)
    for j in range(len(nums) - 1, 0, -1):
        print(nums[j])

factors2(32)